enhancements/microshift/microshift-supportability.md

```diff
@@ -0,0 +1,384 @@
+are the same, so the contents of the plugin may be used too.
+
+Apart from other components, that will be handled by their own plugins (full
+list in a system can be obtained by running `sos report -l`), we need to
+retrieve:
+1. Control plane component logs: apiserver, etcd, kubelet, etc.
+1. Yaml output for all the manifests.
+1. Logs for every container.
+1. MicroShift's configuration and version.
+
+Retrieving all this information involves using a kubernetes client. While
+MicroShift itself uses the go client extensively, the idea of introducing
+another command to produce this output and avoid installing `oc`/`kubectl` is
+not the best, since we would be adding runtime dependencies between a system
+component and MicroShift.
+
+### Workflow Description
+
+> _Disclaimer:_ Workflows described here illustrate manual processes.
+
+Assuming from now on MicroShift has already been installed in a R4E/RHEL based
+system and there has been some kind of issue.
+
+1. MicroShift administrator notices an issue/wants to get a system report.
+1. The administrator will log in to the failing host.
+1. The administrator will perform the debug report creation procedure:
+```
+sudo sos report
+```
+
+For the CI, the same procedure applies.
+
+#### Variation [optional]
+N/A
+
+### API Extensions
+N/A
+
+### Implementation Details/Notes/Constraints [optional]
+N/A
+
+### Risks and Mitigations
+There are three main risks:
+1. Upgrades and incompatible changes.
+1. `sos` release cadence.
+1. Extra required disk space.
+
+#### Upgrades
+When upgrading a MicroShift system we may need to upgrade just MicroShift, or
+the OS as well. If it is the latter then `sos` will be updated before
+MicroShift, therefore we must ensure backwards compatibility with the plugin.
+
+If it is the former then we need to ensure newer MicroShift versions do not
+introduce incompatible changes with the reporting capabilities of the `sos`
+plugin. This may also be seen from a different perspective: the plugin should
+make no assumptions about names or locations of components whenever possible.
+In other words, the plugin must be written in the most generic way that still
+allows retrieval of relevant information.
+
+Still, if incompatible changes are introduced then we need to upgrade `sos` as
+well.
+
+#### Sos release cadence
+`sos` release cadence has been shortened to 4 weeks starting on Feb 1st 2023.
+This change has been introduced to allow faster support of new components and
+features. Since MicroShift follows the same release calendar than OpenShift,
+`sos` is faster.
+
+New versions of `sos` will simply slot in to the next regular release. If there
+is an urgent need a hotfix can be made between releases.
+
+#### Extra disk space
+Adding a plugin to `sos` and not relying on MicroShift binary means it needs
+the `oc` or `kubectl` binary installed. This package is available in the RHEL
+repos, but it comes at an extra cost of 125MB once installed.
```

Comment:
We're planning to introduce dependency on openshift-clients package due to its requirement by Greenboot script.

Reply:
Updated to add the greenboot comment.